service_configuration_lib/spark_config.py

```diff
@@ -163,6 +169,27 @@ def _get_mesos_docker_volumes_conf(
     return {'spark.mesos.executor.docker.volumes': volume_str}
 
 
+def _get_k8s_docker_volumes_conf(
+    volumes: Optional[List[Mapping[str, str]]] = None,
+):
+    env = {}
+    k8s_volumes = volumes or []
+    k8s_volumes.append(DEFAULT_SPARK_K8S_SECRET_VOLUME)
+    for volume_name, volume in enumerate(k8s_volumes):
+        env[f'spark.kubernetes.executor.volumes.hostPath.{volume_name}.mount.path'] = volume['containerPath']
+        env[f'spark.kubernetes.executor.volumes.hostPath.{volume_name}.mount.readOnly'] = (
+            'true' if volume['mode'].lower() == 'ro' else 'false'
+        )
+        env[f'spark.kubernetes.executor.volumes.hostPath.{volume_name}.options.path'] = volume['hostPath']
+    # docker.parameters user needs /etc/passwd and /etc/group to be mounted
+    for i, required in enumerate(['/etc/passwd', '/etc/group']):
+        volume_name = len(k8s_volumes) + i
```

Comment:
This is not great, hence the WIP

Reply:
+1 for just these 2 variables, I would just add it manually. Looks less complicated.
What was the error you were getting if this is not mounted?

Reply:
Or make it similar to what is being done for DEFAULT_SPARK_K8S_SECRET_VOLUME